src/model/document.ts

```diff
@@ -48,6 +48,21 @@ export class ScannedDocument {
   get url() {
     return this.document.url;
   }
+
+  getNestedFeatures(): ScannedFeature[] {
```

Comment:
Refactored into a public method that takes no arguments and a private method that works like this